### ISSUE TITLE
cannot read repositories

### ISSUE THREAD

**Issue by skord** [groupon/DotCi#132]
Platform
* Ubuntu 14.04
* Mongo 3 something
* Jenkins 1.596
* AWS Cloud plugin for build slaves running similar platform
* See the same error on public github, but trying against github enterprise
* DotCI 2.7.3

I have a pretty generic setup. Jenkins gets past the point of parsing the .ci.yml file, but when it goes to clone, it's using what to me seems like an odd choice of remotes and fails. I've not done anything special outside following the installation instructions. 

```
Building remotely on test_runner_pro (i-b2fe6764) (test_runner_pro) in workspace workspace/mdanko/fubar
[fubar] $ /bin/bash -le /tmp/hudson4363158287866233363.sh
$  install_packages ruby-1.9.3_p392
$  export TZ=UTC
$  export LANG=en_US.UTF-8
$  export LC_ALL=en_US.UTF-8
$  export DISPLAY=:1
$  export DOTCI_SHA_SHORT=${DOTCI_SHA:0:10}
$  export RACK_ENV=test
$  export RAILS_ENV=test
$  export BUNDLE_PATH=/tmp/`echo workspace/mdanko/fubar | awk -F "/" '{print $5"/"$6}'`/`jruby -v 2>&1 | awk '{print $1"-"$2}' | grep jruby || ruby -v | awk '{print $1"-"$2}'`/bundle_cache
$  export BUNDLE_FROZEN=true
$  export BUNDLE_WITHOUT=production:development
$  export GEM_HOME=/tmp
$  export PATH=$GEM_HOME/bin:$PATH
$  find . ! -path "./deploykey_rsa.pub" ! -path "./deploykey_rsa" -delete
$  git init
Initialized empty Git repository in /home/jenkins/workspace/mdanko/fubar/.git/
$  git remote add origin git@git.ourdomain.com/api/v3/repos:mdanko/fubar.git
$  git fetch origin master
fatal: 'git@git.ourdomain.com/api/v3/repos:mdanko/fubar.git' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Build step 'Execute shell' marked build as failure
Sending  email notifications
setting commit status on Github for https://git.ourdomain.com/api/v3/repos/mdanko/fubar/commit/cf92b404f867ed26431f0864571d95e45d21e43d
Finished: FAILURE
```

**Comment by suryagaddipati** [comment]
@skord does clone work if you do it manually on that machine ?

**Comment by skord** [comment]
No, I ssh'd to the build slave and it did not work. I tried running it on the Jenkins box itself and it also does not work. 




It seems kind of weird since it can find the .ci.yml file initially (if it's missing I get a report that it is missing), but then the remote is added and that repo does not work. 




Since the remote is a git protocol, I'd expect a deploy key, but there doesn't appear to be one on the build slave.

**Comment by suryagaddipati** [comment]
@skord maybe generate a keypair on the box and add it to a github account?

**Comment by skord** [comment]
I've tried this remote locally and remotely (on Jenkins and a slave) with both a key and an agent with no luck. 

Here's the remote that's automatically added:
```
git@git.ourdomain.com/api/v3/repos:mdanko/fubar.git
```

I'm unsure if that's valid. What I expect would be a valid remote:

```
git@git.ourdomain.com/mdanko/fubar.git
```

Indeed, that's the one I can use, but I don't even see where I'd set that, here's the generated job's config:

![fubar_config__jenkins_](https://cloud.githubusercontent.com/assets/26522/7567540/a33818bc-f7cb-11e4-86a1-464f57693811.png)

I'm assuming those are somehow contrived from the base config:

![configure_system__jenkins_](https://cloud.githubusercontent.com/assets/26522/7567568/dab86bd4-f7cb-11e4-90a1-143ba2a29c0a.png)

**Comment by suryagaddipati** [comment]
hey @skord yea that `Associate Github Repository` seems wrong.  Would you be able to manually change it to the right url( weburl not api url) in the job configuration page and see if that works.

I'll investigate why it configured it to api url for the job.

**Comment by suryagaddipati** [comment]
Fixed in [2.10.0](http://maven.jenkins-ci.org:8081/content/repositories/releases/com/groupon/jenkins-ci/plugins/DotCi/2.10.0/ )

You might have blow away existing projects.

**Comment by peetasan** [comment]
My issue with this was that the in the global configuration `Clone Url Template` under `Docker Compose Build Type` was `https://<DOMAIN>/<ORG>/<REPO>.git` and after an upgrade of DotCI plugin from 2.11.2 to 2.28.1 the remote added had .git.git at the end. Removing the `.git` setting from the config solved the issue.
